source/ui/x11/X11Window.ooc

```diff
@@ -52,8 +64,33 @@ X11Window: class extends NativeWindow {
 	}
 	resize: func (size: IntSize2D) { XResizeWindow(this display, this backend, size width, size height) }
 	free: override func {
+		if (this _xImage)
+			XDestroyImage(this _xImage)
+		XFreeGC(this display, this _x11GraphicsContext)
+		XDestroyWindow(this display, this backend)
 		XCloseDisplay(this display)
 		super()
 	}
+	draw: func (image: RasterYuv420Semiplanar) {
+		if (this _cacheSize != image size) {
+			if (this _xImage)
+				XDestroyImage(this _xImage)
+			imageData := gc_malloc(image width * image height * 4)
+			this _xImage = XCreateImage(this display, CopyFromParent, this _displayDepth, ZPixmap, 0, imageData, image width, image height, 8, 0)
```

Comment:
It should be the value of `bytesPerLine` for the `imageData` array up there, not the `image` object we are trying to display. It is allocated with size `image width * image height * 4` because we can't assume that every X11 system is using 3 bytes per pixel. If you alloc it as `image width * image height * 3` then it will crash on some systems (for example my linux at home). So I don't know it it actually needs to be `image width * 3` or `image width * 4`. The way it is done now you let the Xlib handle this.